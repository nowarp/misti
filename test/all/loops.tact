fun test1(): Int {
  let sum: Int = 0;
  let i: Int = 0;
  while (i < 10) {
    i = i + 1;
    sum = sum + i;
  }
  return sum;
}

fun test2(): Int {
  let sum: Int = 0;
  let i: Int = 0;
  do {
    i = i + 1;
    sum = sum + i;
  } until (i < 10);
  return sum;
}

fun test3(): Int {
  let sum: Int = 0;
  let i: Int = 0;
  repeat (10) {
    i = i + 1;
    sum = sum + i;
  }
  return sum;
}

fun test4() {
    let m: map<Int, Int> = emptyMap();
    foreach (k, v in m) {
    }
}
