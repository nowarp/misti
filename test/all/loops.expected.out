[INFO] PreferAugmentedAssign: Prefer augmented assignment: i += 1
test/all/loops.tact:5:5:
  4 |   while (i < 10) {
> 5 |     i = i + 1;
          ^
  6 |     sum = sum + i;
Help: Consider using augmented assignment instead: i += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: sum += i
test/all/loops.tact:6:5:
  5 |     i = i + 1;
> 6 |     sum = sum + i;
          ^
  7 |   }
Help: Consider using augmented assignment instead: sum += i
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: i += 1
test/all/loops.tact:15:5:
  14 |   do {
> 15 |     i = i + 1;
           ^
  16 |     sum = sum + i;
Help: Consider using augmented assignment instead: i += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: sum += i
test/all/loops.tact:16:5:
  15 |     i = i + 1;
> 16 |     sum = sum + i;
           ^
  17 |   } until (i < 10);
Help: Consider using augmented assignment instead: sum += i
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: i += 1
test/all/loops.tact:25:5:
  24 |   repeat (10) {
> 25 |     i = i + 1;
           ^
  26 |     sum = sum + i;
Help: Consider using augmented assignment instead: i += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: sum += i
test/all/loops.tact:26:5:
  25 |     i = i + 1;
> 26 |     sum = sum + i;
           ^
  27 |   }
Help: Consider using augmented assignment instead: sum += i
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign